enterprise_access/apps/bffs/tests/test_context.py

```diff
@@ -1,38 +1,80 @@
-from enterprise_access.utils import _curr_date
 
 
+@ddt.ddt
 class TestHandlerContext(TestCase):
     def setUp(self):
         super().setUp()
         self.factory = RequestFactory()
         self.mock_user = UserFactory()
+        self.faker = Faker()
+
+        self.mock_enterprise_customer_uuid = self.faker.uuid4()
+        self.request = self.factory.get('sample/api/call')
+        self.request.user = self.mock_user
+        self.request.query_params = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.request.data = {
+            'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
+        }
+        self.context = HandlerContext(self.request)
 
     def test_handler_context_init(self):
-        request = self.factory.get('sample/api/call')
-        request.user = self.mock_user
-        context = HandlerContext(request)
+        context = HandlerContext(self.request)
 
-        self.assertEqual(context.request, request)
+        self.assertEqual(context.request, self.request)
         self.assertEqual(context.user, self.mock_user)
         self.assertEqual(context.data, {})
         self.assertEqual(context.errors, [])
         self.assertEqual(context.warnings, [])
-        self.assertEqual(context.enterprise_customer_uuid, None)
-        self.assertEqual(context.lms_user_id, None)
+        self.assertEqual(context.enterprise_customer_uuid, self.mock_enterprise_customer_uuid)
+        self.assertEqual(context.lms_user_id, self.mock_user.lms_user_id)
+
+    @ddt.data(
+        {
+            'query_params': True,
+            'data': True,
+        },
+        {
+            'query_params': False,
+            'data': True,
+        },
+        {
+            'query_params': True,
+            'data': False,
+        },
+        {
+            'query_params': False,
+            'data': False,
+        }
+    )
+    @ddt.unpack
+    def test_handler_context_enterprise_customer_uuid(self, query_params, data):
+        if not query_params:
+            del self.request.query_params['enterprise_customer_uuid']
+        if not data:
+            del self.request.data['enterprise_customer_uuid']
+        request = self.request
+        if not (query_params or data):
+            with self.assertRaises(ValueError):
+                HandlerContext(request)
+        else:
+            context = HandlerContext(request)
+            self.assertEqual(context.enterprise_customer_uuid, self.mock_enterprise_customer_uuid)
+            self.assertEqual(context.lms_user_id, self.mock_user.lms_user_id)
 
     def test_handler_context_add_error_serializer(self):
-        request = self.factory.get('sample/api/call')
-        request.user = self.mock_user
-        context = HandlerContext(request)
+        context = self.context
         expected_output = {
             "developer_message": "No enterprise uuid associated to the user mock-uuid",
```

Comment:
[curious] Where is `mock-uuid` coming from? Typically the user id is an integer, not a uuid so having this (and a few other places below) state its a uuid could be misleading?